src/bp-core/compatibility/bp-incompatible-plugins-helper.php

```diff
@@ -562,21 +562,46 @@ function bp_core_add_support_for_google_captcha_pro( $section_notice, $section_s
 
 add_filter( 'gglcptch_section_notice', 'bp_core_add_support_for_google_captcha_pro', 100, 2 );
 
-
 /**
  * Update the BuddyBoss Platform Fields when user register from MemberPress Registration form
  *
  * Support MemberPress and MemberPress Pro
  *
  * @since BuddyBoss 1.1.9
  */
-function bp_core_add_support_mepr_signup_map_user_fields( $txn ) {
+function bb_core_add_support_mepr_signup_map_user_fields( $txn ) {
 	if ( ! empty( $txn->user_id ) ) {
 		bp_core_map_user_registration( $txn->user_id, true );
 	}
 }
 
-add_action( 'mepr-signup', 'bp_core_add_support_mepr_signup_map_user_fields', 100 );
+add_action( 'mepr-signup', 'bb_core_add_support_mepr_signup_map_user_fields', 100 );
+
+/**
+ * Prevent MemberPress registration when nickname(username) format is not valid
+ * 
+ * @since BuddyBoss [BBVERSION]
+ *
+ * @param array $errors		Array of error messages from memberpress signup validation
+ * 
+ * @return array $errors	Array of error messages from memberpress signup validation
+ */
+function bb_core_validate_nickname_mepr_signup( $errors ) {
+	if ( function_exists( 'bp_xprofile_nickname_field_id' ) ) {
+		$nickname   = '';
+		if ( isset( $_POST['user_login'] ) ) {
+		$nickname = sanitize_text_field( $_POST['user_login'] );
+		}
+		$field_id   = bp_xprofile_nickname_field_id();
+		$message    = bp_xprofile_validate_nickname_value( '', $field_id, $nickname, '' );
+		if ( ! empty( $message ) ) {
+		$errors['user_login'] =  $message;
+		}
+	}
+	return $errors;
+}
```

Comment:
@mahedihasannoman please follow the phpcs here.

Reply:
Done